rbi/core/time.rbi

```diff
@@ -355,13 +355,13 @@ class Time < Object
   # ```
   sig do
     params(
-        arg0: Time,
+        arg0: BasicObject,
```

Comment:
I guess this specific overload still makes sense, since if a `Time` instance is supplied, the result would be a `Float` but otherwise, the value would be coerced to a numeric value and the result would be a `Time.`